core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -504,16 +510,23 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
-    private void registerBuiltinFunctionScoreParsers() {
-        registerFunctionScoreParser(new ScriptScoreFunctionParser());
-        registerFunctionScoreParser(new GaussDecayFunctionParser());
-        registerFunctionScoreParser(new LinearDecayFunctionParser());
-        registerFunctionScoreParser(new ExponentialDecayFunctionParser());
-        registerFunctionScoreParser(new RandomScoreFunctionParser());
-        registerFunctionScoreParser(new FieldValueFactorFunctionParser());
+    private void registerBuiltinScoreFunctionParsers() {
+        registerScoreFunction(ScriptScoreFunctionBuilder.FIELD, ScriptScoreFunctionBuilder::fromXContent, ScriptScoreFunctionBuilder.NAME,
```

Comment:
shall we settle in some naming convention for
1) the field name: we used QUERY_NAME_FIELD for queries and are introducing FIELD here. Shall we use FUNCTION_NAME_FIELD instead? Too long?
2) More importantly, the order of arguments for the register method, that is exposed to plugins. I have the field last, but it does make sense to have it first I guess, is this what we want to do everywhere?